packages/client/tests/functional/runtime-import/tests.ts

```diff
@@ -16,24 +15,24 @@ testMatrix.setupTestSuite((suiteConfig, suiteMeta, clientMeta) => {
 
     if (clientMeta.runtime === 'edge') {
       expect(generatedClientContents).toContain(edgeRuntime)
-      expect(generatedClientContents).not.toContain(dataProxyRuntime)
       expect(generatedClientContents).not.toContain(libraryRuntime)
       expect(generatedClientContents).not.toContain(binaryRuntime)
-    } else if (clientMeta.dataProxy) {
-      expect(generatedClientContents).toContain(dataProxyRuntime)
+    } else if (clientMeta.dataProxy && getClientEngineType() === ClientEngineType.Library) {
+      expect(generatedClientContents).toContain(libraryRuntime)
+      expect(generatedClientContents).not.toContain(edgeRuntime)
       expect(generatedClientContents).not.toContain(binaryRuntime)
-      expect(generatedClientContents).not.toContain(libraryRuntime)
+    } else if (clientMeta.dataProxy && getClientEngineType() === ClientEngineType.Binary) {
+      expect(generatedClientContents).toContain(binaryRuntime)
       expect(generatedClientContents).not.toContain(edgeRuntime)
+      expect(generatedClientContents).not.toContain(libraryRuntime)
     } else if (getClientEngineType() === ClientEngineType.Library) {
       expect(generatedClientContents).toContain(libraryRuntime)
-      expect(generatedClientContents).not.toContain(binaryRuntime)
-      expect(generatedClientContents).not.toContain(dataProxyRuntime)
       expect(generatedClientContents).not.toContain(edgeRuntime)
+      expect(generatedClientContents).not.toContain(binaryRuntime)
     } else if (getClientEngineType() === ClientEngineType.Binary) {
+      expect(generatedClientContents).toContain(edgeRuntime)
```

Comment:
```suggestion
      expect(generatedClientContents).not.toContain(edgeRuntime)
```
Would make more sense here, like it's done with Library?

Reply:
Yes, I agree. It was alwso caught by the tests.